api/indexer.oas2.json

```diff
@@ -485,6 +485,94 @@
         }
       }
     },
+    "/v2/applications/{application-id}/boxes": {
+      "get": {
+        "description": "Given an application ID, returns the box names of that application sorted lexicographically.",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "tags": [
+          "search"
+        ],
+        "operationId": "searchForApplicationBoxes",
+        "summary": "Get box names for a given application.",
+        "parameters": [
+          {
+            "type": "integer",
+            "name": "application-id",
+            "in": "path",
+            "required": true
+          },
+          {
+            "$ref": "#/parameters/limit"
+          },
+          {
+            "$ref": "#/parameters/next"
+          }
+        ],
+        "responses": {
+          "200": {
+            "$ref": "#/responses/BoxesResponse"
+          },
+          "400": {
+            "$ref": "#/responses/ErrorResponse"
+          },
+          "404": {
+            "$ref": "#/responses/ErrorResponse"
+          },
+          "500": {
+            "$ref": "#/responses/ErrorResponse"
+          }
+        }
+      }
+    },
+    "/v2/applications/{application-id}/box": {
```

Comment:
I think this would fit with the existing endpoints a little better as:
`/v2/applications/{application-id}/boxes/{box-name}`

I believe the spec is a little awkward with parameters that appear on the path and as query params, you can look at how we deal with that elsewhere by looking at how the address is handled on `/v2/transactions?account` and `/v2/accounts/{account-id}`

Reply:
It sounds like the proposed change would make boxes work more like other endpoints in `indexer`. However, then it will diverge from the usage in `algod` which I was aiming to keep unified. Another downside is that the SDK work has already been completed for ingesting these new endpoints. However, if the user experience is adversely affected, or it makes developing against indexer's API too burdensome, we should make the change.

Reply:
Do you know why it was changed? It appears that it used the format I suggested in the original implementation:
https://github.com/algorand/go-algorand/issues/3973

Reply:
The path indicates that it is a required parameter.

The SDKs would also generate a more intuitive interface. For example:

With the query parameter, the request looks like this:
```
// this is illegal, but allowed.
var badBoxResp = client.lookupApplicationBoxByIDAndName().execute();

// this is legal.
var boxResp = client.lookupApplicationBoxByIDAndName().BoxName("mybox").execute();
```

With a path parameter, it looks like this:
```
var boxResp = client.lookupApplicationBoxByIDAndName("mybox").execute();
```

Reply:
It just occurred to me that if it must be a query parameter, why not have a single `/v2/applications/{application-id}/boxes` endpoint with an optional name filter?

Reply:
@winder Responding inline below to provide context.  It might be easier to discuss verbally, happy to do so.

> Do you know why it was changed?

* The team went through several iterations prior to landing on the goal app call encoding scheme to safely represent binary data.
* Somewhat subjectively and I can elaborate live:  I preferred a solution using query string parameters over path segments because I think URL encoding query string parameters is less likely to be implemented incorrectly in SDKs (ours or otherwise). 
* In retrospect, I should have raised the API deviation with a broader group sooner.  That's my fault - sorry for that.  I'll obviously be more mindful moving ahead.
* Given that we're already here, my personal preference is to keep as is with the understanding that we're _not_ aiming to propagate the convention elsewhere.  If you feel strongly about retaining existing conventions, I'd prefer to move to a live discussion.

>  why not have a single /v2/applications/{application-id}/boxes endpoint with an optional name filter?

I don't know if it's convincing, but here's a rationale:  The 2 endpoints in-question have different return types.  Keeping distinct endpoints makes the return type explicit + unconditional.

Reply:
Resolving as the discussion is now happening elsewhere.